Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/contactdiary/ContactDiaryDayFragmentTest.kt

```diff
@@ -0,0 +1,184 @@
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import de.rki.coronawarnapp.contactdiary.ui.day.ContactDiaryDayFragment
+import de.rki.coronawarnapp.contactdiary.ui.day.ContactDiaryDayFragmentArgs
+import de.rki.coronawarnapp.contactdiary.ui.day.ContactDiaryDayViewModel
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.location.ContactDiaryLocationListFragment
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.location.ContactDiaryLocationListViewModel
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.person.ContactDiaryPersonListFragment
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.person.ContactDiaryPersonListViewModel
+import de.rki.coronawarnapp.contactdiary.util.SelectableItem
+import de.rki.coronawarnapp.contactdiary.util.toFormattedDay
+import de.rki.coronawarnapp.ui.contactdiary.DiaryData.LOCATIONS
+import de.rki.coronawarnapp.ui.contactdiary.DiaryData.PERSONS
+import io.mockk.MockKAnnotations
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.spyk
+import org.joda.time.LocalDate
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import testhelpers.BaseUITest
+import testhelpers.SCREENSHOT_DELAY_TIME
+import testhelpers.Screenshot
+import testhelpers.SystemUIDemoModeRule
+import testhelpers.TestDispatcherProvider
+import testhelpers.launchFragment2
+import testhelpers.launchFragmentInContainer2
+import testhelpers.selectTabAtPosition
+import tools.fastlane.screengrab.Screengrab
+import tools.fastlane.screengrab.locale.LocaleTestRule
+import tools.fastlane.screengrab.locale.LocaleUtil
+import java.util.Locale
+
+@RunWith(AndroidJUnit4::class)
+class ContactDiaryDayFragmentTest : BaseUITest() {
+
+    @Rule
+    @JvmField
+    val localeTestRule = LocaleTestRule()
+
+    @get:Rule
+    val systemUIDemoModeRule = SystemUIDemoModeRule()
+
+    @MockK lateinit var contactDiaryRepository: ContactDiaryRepository
+
+    private lateinit var viewModel: ContactDiaryDayViewModel
+    private lateinit var personListViewModel: ContactDiaryPersonListViewModel
+    private lateinit var locationListViewModel: ContactDiaryLocationListViewModel
+
+    private val selectedDay = LocalDate.now().toString()
+    private val fragmentArgs = ContactDiaryDayFragmentArgs(selectedDay = selectedDay).toBundle()
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this, relaxed = true)
+        setupViewModels()
+
+        every { viewModel.uiState } returns MutableLiveData(
+            ContactDiaryDayViewModel.UIState(
+                dayText = { LocalDate.now().toFormattedDay(LocaleUtil.getTestLocale() ?: Locale.getDefault()) },
+                dayTextContentDescription = { "Description" }
+            )
+        )
+    }
+
+    @After
+    fun teardown() {
+        clearAllViewModels()
+    }
+
+    @Test
+    fun launch_fragment() {
```

Comment:
a very minor thing for the future: in unit tests, we have the convention of naming tests as follows
`fun `launch fragment()``